### PR TITLE
create initial backend and rest api function

### PR DESCRIPTION
**Pull request by Nasser-Eddin-Nasser**

### CODE REVIEW

**Comment by MBuchalik** [comment]
@Nasser-Eddin-Nasser Is the PR template only meant for the backend devs, or also for the frontend devs? I am asking because we frontend devs will most likely need another template (if we even need one)... Maybe, you could label the template as "for backend PRs only"?

File: .github/merge_request_template/Feature.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Merge request Check-list:
```
Comment (MBuchalik) [inline .github/merge_request_template/Feature.md]: On GitHub, you say "Pull Request", not "Merge Request" :)

Comment (Nasser-Eddin-Nasser) [reply .github/merge_request_template/Feature.md]: Hey @MBuchalik , thanks for your notes. Yes, it's only for the backends devs. I'll label it to Backend.

**Comment by MBuchalik** [comment]
This PR is still not ready to be merged, am I right? In this case, it could be useful if you mark the PR as a draft. This ensures that nobody accidentally merges it.
When creating a PR, you can mark it as a draft like so: https://github.blog/2019-02-14-introducing-draft-pull-requests/
If you created a non-draft PR and want to convert it to a draft: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft

I haven't yet used this feature, so I am not sure about the implications of it. But I am pretty sure it should just work as expected.

**Review by dalex135 — APPROVED** [review]

**Review by vishwasanavatti — APPROVED** [review]

**Comment by vishwasanavatti** [comment]
compiled and ran successfully. The basic flow worked just fine. pushed to master to release it for tthe current spint.

**Pull request merged**